gapic-generator/templates/default/service/client/_resource.erb

```diff
@@ -7,19 +7,17 @@
 # `<%= resource.patterns.first.template %>`
 #
 <%- resource.patterns.first.arguments.each do |arg| -%>
-# @param <%= arg.name %> [String]
-<%- if arg.desc -%>
-#   <%= arg.desc %>
-<%- end -%>
+# @param <%= arg %> [String]
 <%- end -%>
 #
 # @return [String]
-<%- args = resource.patterns.first.arguments.map { |arg| "#{arg.name}:" } -%>
-def <%= resource.path_helper %> <%= args.join ", " %>
-<%- resource.patterns.first.arguments.each do |arg| -%>
-  raise ArgumentError, "<%= arg.name %> is required" if <%= arg.name %>.nil?
-<%- if arg.regexp -%>
-  raise ArgumentError, <%= arg.msg.inspect %> if <%= arg.regexp %>.match? <%= arg.name %>
+<%- args = resource.patterns.first.arguments.map { |arg| "#{arg}:" }.join ", " -%>
+def <%= resource.path_helper %> <%= args %>
+<%- last_argument_index = resource.patterns.first.arguments.count - 1 -%>
+<%- resource.patterns.first.arguments.each_with_index do |arg, index| -%>
+  raise ArgumentError, "<%= arg %> must be a String" unless <%= arg %>.is_a? String
```

Comment:
This is probably not necessarily true either. It could be something with a reasonable `to_s` like an Integer. I still think this particular check is redundant. I'd suggest removing it (and updating the other one to check for `<%= arg %>.to_s.include? "/"`)

Reply:
Okay, I'll make the change.

Reply:
Updated. PTAL.